source/plugins/ruby/podinventory_to_mdm.rb

```diff
@@ -93,67 +93,71 @@ def initialize(custom_metrics_azure_regions)
   end
 
   def get_pod_inventory_mdm_records(batch_time)
+    records = []
     begin
-      # generate all possible values of non_phase_dim_values X pod Phases and zero-fill the ones that are not already present
-      @no_phase_dim_values_hash.each { |key, value|
-        @@pod_phase_values.each { |phase|
-          pod_key = [key, phase].join("~~")
-          if !@pod_count_hash.key?(pod_key)
-            @pod_count_hash[pod_key] = 0
-          else
+      if @process_incoming_stream
+        # generate all possible values of non_phase_dim_values X pod Phases and zero-fill the ones that are not already present
+        @no_phase_dim_values_hash.each { |key, value|
+          @@pod_phase_values.each { |phase|
+            pod_key = [key, phase].join("~~")
+            if !@pod_count_hash.key?(pod_key)
+              @pod_count_hash[pod_key] = 0
+            else
+              next
+            end
+          }
+        }
+        @pod_count_hash.each { |key, value|
+          key_elements = key.split("~~")
+          if key_elements.length != 4
             next
           end
-        }
-      }
-      records = []
-      @pod_count_hash.each { |key, value|
-        key_elements = key.split("~~")
-        if key_elements.length != 4
-          next
-        end
 
-        # get dimension values by key
-        podNodeDimValue = key_elements[0]
-        podNamespaceDimValue = key_elements[1]
-        podControllerNameDimValue = key_elements[2]
-        podPhaseDimValue = key_elements[3]
+          # get dimension values by key
```

Comment:
bunch whitespaces and i think, we need to get rid off these to have cleaner code. I use this setting in vscode "files.trimTrailingWhitespace": true to avoid getting these whitespaces

Reply:
this is coming from formatter. I have ruby formatter that is formatting the code